Sources/KeyboardShortcuts/Recorder.swift

```diff
@@ -31,39 +31,70 @@ extension KeyboardShortcuts {
 
 		private let name: Name
 		private let onChange: ((_ shortcut: Shortcut?) -> Void)?
+        private let isHorizontalContentSizeConstraintActive: Bool
 
-		/**
-		- Parameter name: Strongly-typed keyboard shortcut name.
-		- Parameter onChange: Callback which will be called when the keyboard shortcut is changed/removed by the user. This can be useful when you need more control. For example, when migrating from a different keyboard shortcut solution and you need to store the keyboard shortcut somewhere yourself instead of relying on the built-in storage. However, it's strongly recommended to just rely on the built-in storage when possible.
-		*/
+        /**
+         - Parameter name: Strongly-typed keyboard shortcut name.
+         - Parameter onChange: Callback which will be called when the keyboard shortcut is changed/removed by the user. This can be useful when you need more control. For example, when migrating from a different keyboard shortcut solution and you need to store the keyboard shortcut somewhere yourself instead of relying on the built-in storage. However, it's strongly recommended to just rely on the built-in storage when possible.
+         - Parameter isHorizontalContentSizeConstraintActive: When `true`, makes sure to have width constraint set to avoid cutting placeholder string. When `false`, the frame(:) modifier will control the actual width of the text field.
+         */
 		public init(
 			for name: Name,
-			onChange: ((_ shortcut: Shortcut?) -> Void)? = nil
+			onChange: ((_ shortcut: Shortcut?) -> Void)? = nil,
+            constraint isHorizontalContentSizeConstraintActive: Bool = false
 		) {
 			self.name = name
 			self.onChange = onChange
+            self.isHorizontalContentSizeConstraintActive = isHorizontalContentSizeConstraintActive
 		}
 
 		/// :nodoc:
-		public func makeNSView(context: Context) -> NSViewType { .init(for: name, onChange: onChange) }
+		public func makeNSView(context: Context) -> NSViewType { .init(for: name, onChange: onChange, isHorizontalContentSizeConstraintActive: isHorizontalContentSizeConstraintActive) }
 
 		/// :nodoc:
 		public func updateNSView(_ nsView: NSViewType, context: Context) {
 			nsView.shortcutName = name
+            
+            // support for controlSize(:) modifier
+            switch context.environment.controlSize {
+                case .small:
+                    nsView.font = .systemFont(ofSize: 11)
+                    nsView.changeWidth(width: 110)
```

Comment:
Why manually apply this instead of just using `NSControl#controlSize`?

Reply:
Good point, will fix it